src/sc_stage.py

```diff
@@ -181,11 +184,12 @@ def stop(self):
         # stop the production loop
         self.stage_active.clear()
 
-        # clear in-memory state
-        self.outbound_material = None
+        # clear in-memory state;
+        # Init to Queue() as it sometimes fails due to race condition
+        self.outbound_material = Queue()
         for item_type in self.inbound_material:
             # Set all inbound queues to null
-            self.inbound_material[item_type] = None
+            self.inbound_material[item_type] = Queue()
```

Comment:
@raokrutarth setting this to Queue() instead of None. There are race conditions between inactive and  put(). Verify if this works.